kac-api/src/utils/validation-server.ts

```diff
@@ -0,0 +1,173 @@
+import fastify from "fastify";
+import pino from 'pino';
+import path from "path";
+import {readFileSync} from "fs";
+import {MonokleValidator, Resource} from "@monokle/validation";
+import {V1ValidatingWebhookConfiguration, V1ObjectMeta} from "@kubernetes/client-node";
+
+export type ValidationServerOptions = {
+  port: number;
+  host: string;
+};
+
+export type AdmissionRequest = V1ValidatingWebhookConfiguration & {
+  request?: V1ObjectMeta & {
+    object?: Resource
+  }
+};
+
+export type AdmissionResponse = {
+  kind: string,
+  apiVersion: string,
+  response: {
+      uid: string,
+      allowed: boolean,
+      status: {
+          message: string
+      }
+  }
+};
+
+export class ValidationServer {
+  private _server: ReturnType<typeof fastify>;
+  private _shouldValidate: boolean
+
+  constructor(
+    private readonly _validator: MonokleValidator,
+    private readonly _logger: ReturnType<typeof pino>,
+    private readonly _options: ValidationServerOptions = {
+      port: 8443,
+      host: '0.0.0.0'
+    }
+  ) {
+    this._shouldValidate = false;
+
+    this._server = fastify({
+      // @TODO do not require certs when running locally (for testing outside K8s cluster)
+      https: {
+        key: readFileSync(path.join('/run/secrets/tls', 'tls.key')),
+        cert: readFileSync(path.join('/run/secrets/tls', 'tls.crt'))
+      }
+    });
+
+    this._initRouting();
+  }
+
+  get shouldValidate() {
+    return this._shouldValidate;
+  }
+
+  set shouldValidate(value: boolean) {
+    this._shouldValidate = value;
+  }
+
+  async start() {
+    return new Promise((resolve, reject) => {
+      this._server.listen({port: this._options.port, host: this._options.host}, (err, address) => {
+        if (err) {
+          reject(err);
+        }
+
+        this._logger.info(`Server listening at ${address}`);
+
+        this.shouldValidate = true;
+
+        resolve(address);
+      });
+
+    });
+  }
+
+  async stop() {
+    this.shouldValidate = false;
+
+    if (this._server) {
+      await this._server.close();
+    }
+  }
+
+  private async _initRouting() {
+    this._server.post("/validate", async (req, _res): Promise<AdmissionResponse> => {
+
+      this._logger.debug({request: req})
+      this._logger.trace({requestBody: req.body});
+
+      const body = req.body as AdmissionRequest;
+
+      const response = {
+        kind: body?.kind || '',
+        apiVersion: body?.apiVersion || '',
+        response: {
+          uid: body?.request?.uid || "",
+          allowed: true,
+          status: {
+            message: "OK"
+          }
+        }
+      }
+
+      // Dev workaround - always return true for webhook server to not block hot-reload
+      if (body.request?.name?.startsWith('webhook-server-')) {
```

Comment:
I don't like this, I'm not sure of the implications but seems like something that should not get to production. 
Maybe if it's really required / helpful hide it after a development env / build variable?

Reply:
I'm not sure if we still need it, it's a workaround from hackaton still, will recheck this again 👍